Sources/Fullscreen/Consent/Details/ConsentActionView.swift

```diff
@@ -0,0 +1,106 @@
+        scroll.alwaysBounceVertical = true
+        return scroll
+    }()
+
+    // MARK: - Public properties
+
+    public var buttonStyle: Button.Style? {
+        get { return button?.style }
+        set { button = createButton(with: newValue) }
+    }
+
+    public var text: String? {
+        get { return textLabel.attributedText?.string }
+        set { textLabel.attributedText = newValue?.attributedStringWithLineSpacing(4) }
+    }
+
+    public var buttonTitle: String? {
+        didSet { button?.setTitle(buttonTitle, for: .normal) }
+    }
+
+    public var action: (() -> Void)?
+
+    // MARK: - Setup
+
+    public override init(frame: CGRect) {
+        super.init(frame: frame)
+        setupSubviews()
+    }
+
+    public required init?(coder aDecoder: NSCoder) {
+        fatalError("init(coder:) has not been implemented")
+    }
+}
+
+// MARK: - Private functions
+
+private extension ConsentActionView {
+
+    func createButton(with style: Button.Style?) -> Button? {
+        if let button = button {
+            button.removeFromSuperview()
+        }
+
+        guard let style = style else { return nil }
+        let newButton = Button(style: style)
+        newButton.setTitle(buttonTitle, for: .normal)
+        newButton.addTarget(self, action: #selector(handleButtonPressed), for: .touchUpInside)
+        newButton.translatesAutoresizingMaskIntoConstraints = false
+        scrollView.addSubview(newButton)
+
+        let constraints = [
+            newButton.leadingAnchor.constraint(equalTo: leadingAnchor, constant: .mediumLargeSpacing),
+            newButton.trailingAnchor.constraint(equalTo: trailingAnchor, constant: -.mediumLargeSpacing),
+            newButton.topAnchor.constraint(equalTo: textLabel.bottomAnchor, constant: .largeSpacing),
+            scrollView.bottomAnchor.constraint(equalTo: newButton.bottomAnchor, constant: .mediumLargeSpacing)
+        ]
+        NSLayoutConstraint.activate(constraints)
+
+        return button
+    }
+
+    func setupSubviews() {
+        addSubview(scrollView)
+        scrollView.addSubview(textLabel)
+        NSLayoutConstraint.activate([
+            scrollView.leadingAnchor.constraint(equalTo: leadingAnchor),
+            scrollView.topAnchor.constraint(equalTo: topAnchor),
+            scrollView.widthAnchor.constraint(equalTo: widthAnchor),
+            scrollView.heightAnchor.constraint(equalTo: heightAnchor),
+
+            textLabel.leadingAnchor.constraint(equalTo: leadingAnchor, constant: .mediumLargeSpacing),
+            textLabel.topAnchor.constraint(equalTo: scrollView.topAnchor, constant: .mediumLargeSpacing),
+            textLabel.trailingAnchor.constraint(equalTo: trailingAnchor, constant: -.mediumLargeSpacing),
+
+            scrollView.bottomAnchor.constraint(greaterThanOrEqualTo: textLabel.bottomAnchor, constant: .mediumLargeSpacing)
+        ])
+    }
+
+    @objc func handleButtonPressed() {
+        action?()
```

Comment:
Can you switch this to a delegate? I think we've concluded that sticking to delegates for callbacks scales better (since you can just add more method calls) and also makes it easier to maintain consistency.

Reply:
I did that at first, but I would have to add more functionality to make it work. In the FINN-app, this looks pretty good!